docs/source/llms/openai/autologging.rst

```diff
@@ -0,0 +1,184 @@
+    * - Model Signatures
+      - ``false``
+      - ``log_model_signatures``
+      - If set to ``True``, :py:class:`ModelSignatures <mlflow.models.ModelSignature>` describing model inputs and outputs are collected and logged along with OpenAI model artifacts during inference. This option is only available when ``log_models`` is enabled.
+    * - Input Example
+      - ``false``
+      - ``log_input_examples``
+      - If set to ``True``, input examples from inference data are collected and logged along with OpenAI model artifacts during inference. This option is only available when ``log_models`` is enabled.
+    * - Datasets
+      - ``false``
+      - ``log_datasets``
+      - If set to ``True`` and a dataset is used with interacting with the OpenAI model, information about the dataset will be logged.
+
+
+For example, to disable logging of traces, and instead enable model logging, run the following code:
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.openai.autolog(
+        log_traces=False,
+        log_models=True,
+    )
+
+Example of using OpenAI Autologging
+-----------------------------------
+
+.. code-block:: python
+
+    import os
+
+    import mlflow
+    import openai
+
+    OPENAI_MODEL = "gpt-4o"
+    TEMP = 0.95
+    API_KEY = os.environ.get("OPENAI_API_KEY")
+    EXPERIMENT_NAME = "OpenAI Autologging Demonstration"
+    REGISTERED_MODEL_NAME = "openai-auto"
+    MODEL_VERSION = 1
+
+    mlflow.openai.autolog(
+        log_input_examples=True,
+        log_model_signatures=True,
+        log_models=True,
+        log_traces=True,
+        registered_model_name=REGISTERED_MODEL_NAME,
+    )
+
+    mlflow.set_experiment(EXPERIMENT_NAME)
+
+    openai_client = openai.OpenAI(api_key=API_KEY)
+
+    messages = [
+        {
+            "role": "user",
+            "content": "State that you are responding to a test and that you are alive.",
+        }
+    ]
+
+    openai_client.chat.completions.create(
+        model=OPENAI_MODEL,
+        messages=messages,
+        temperature=TEMP,
+    )
+
+Viewing the logged model and the trace used when invoking the OpenAI client within the UI can be seen in the image below:
+
+.. figure:: ../../_static/images/tutorials/llms/openai-autolog.gif
+    :alt: OpenAI Autologging artifacts and traces
+    :width: 100%
+    :align: center
+
+The model can be loaded by using the ``models`` uri via the model that was logged and registered and interfaced with via the pyfunc API as shown below:
+
+.. code-block:: python
+
+    loaded_autologged_model = mlflow.pyfunc.load_model(
+        f"models:/{REGISTERED_MODEL_NAME}/{MODEL_VERSION}"
+    )
+
+    loaded_autologged_model.predict(
+        "How much relative time difference would occur between an astronaut travelling at 0.98c for 14 years "
+        "as measured by an on-board clock on the spacecraft and humans on Earth, assuming constant speed?"
+    )
+
+
+FAQ
+---
+
+How can I log all of my traces while developing?
```

Comment:
do we need to explain this on this page?

Reply:
+1, maybe we can change the title to "How can I manually create trace for OpenAI request?" and link to the fluent API doc? (I guess the intention here was to introduce manual tracing)

Reply:
Great idea! Converted to a reference link. Yes, the intention was to simply inform users who only go to this page that tracing is not solely directly linked to autologging for the OpenAI flavor.